build: Configuring with `-mno-sse4.1` does not fail the sse4.1 instrinsics check

I would have expected that `CXXFLAGS="-mno-sse4.1" ./configure` would make the sse4.1 intrinsics check fail but it doesn't:

```
configure:23377: checking for SSE4.1 intrinsics
configure:23396: clang++ -std=c++17 -c -msse4.1 -mno-sse4.1  conftest.cpp >&5
configure:23396: $? = 0
configure:23398: result: yes
```

Not sure what is going on here, seems like a bug.

Note that this ultimately results in compile failure, but because we try and compile the `sha256_x86_shani.cpp` code, but fail because it `"needs isa option -msse4.1"`:
```bash
In file included from /usr/lib/gcc/x86_64-linux-gnu/13/include/immintrin.h:39,
                 from crypto/sha256_x86_shani.cpp:12:
In function 'void {anonymous}::Shuffle(__m128i&, __m128i&)',
    inlined from 'void sha256_x86_shani::Transform(uint32_t*, const unsigned char*, size_t)' at crypto/sha256_x86_shani.cpp:87:12:
crypto/sha256_x86_shani.cpp:57:10: error: '__builtin_ia32_pblendw128' needs isa option -msse4.1
   57 |     s1 = _mm_blend_epi16(t2, t1, 0xF0);
      |          ^~~~~~~~~~~~~~~
In function 'void {anonymous}::Unshuffle(__m128i&, __m128i&)',
    inlined from 'void sha256_x86_shani::Transform(uint32_t*, const unsigned char*, size_t)' at crypto/sha256_x86_shani.cpp:138:14:
crypto/sha256_x86_shani.cpp:64:10: error: '__builtin_ia32_pblendw128' needs isa option -msse4.1
   64 |     s0 = _mm_blend_epi16(t1, t2, 0xF0);
      |          ^~~~~~~~~~~~~~~
In function 'void {anonymous}::Unshuffle(__m128i&, __m128i&)',
    inlined from 'void sha256d64_x86_shani::Transform_2way(unsigned char*, const unsigned char*)' at crypto/sha256_x86_shani.cpp:270:14:
crypto/sha256_x86_shani.cpp:64:10: error: '__builtin_ia32_pblendw128' needs isa option -msse4.1
   64 |     s0 = _mm_blend_epi16(t1, t2, 0xF0);
      |          ^~~~~~~~~~~~~~~
In function 'void {anonymous}::Unshuffle(__m128i&, __m128i&)',
    inlined from 'void sha256d64_x86_shani::Transform_2way(unsigned char*, const unsigned char*)' at crypto/sha256_x86_shani.cpp:271:14:
crypto/sha256_x86_shani.cpp:64:10: error: '__builtin_ia32_pblendw128' needs isa option -msse4.1
   64 |     s0 = _mm_blend_epi16(t1, t2, 0xF0);
      |          ^~~~~~~~~~~~~~~
In function 'void {anonymous}::Unshuffle(__m128i&, __m128i&)',
    inlined from 'void sha256d64_x86_shani::Transform_2way(unsigned char*, const unsigned char*)' at crypto/sha256_x86_shani.cpp:348:14:
crypto/sha256_x86_shani.cpp:64:10: error: '__builtin_ia32_pblendw128' needs isa option -msse4.1
   64 |     s0 = _mm_blend_epi16(t1, t2, 0xF0);
      |          ^~~~~~~~~~~~~~~
In function 'void {anonymous}::Unshuffle(__m128i&, __m128i&)',
    inlined from 'void sha256d64_x86_shani::Transform_2way(unsigned char*, const unsigned char*)' at crypto/sha256_x86_shani.cpp:349:14:
crypto/sha256_x86_shani.cpp:64:10: error: '__builtin_ia32_pblendw128' needs isa option -msse4.1
   64 |     s0 = _mm_blend_epi16(t1, t2, 0xF0);
      |          ^~~~~~~~~~~~~~~
make[2]: *** [Makefile:14527: crypto/libbitcoin_crypto_x86_shani_la-sha256_x86_shani.lo] Error 1
```
and also because we then try and compile the sse4.1 code anyways:
```bash
In file included from /usr/lib/gcc/x86_64-linux-gnu/13/include/immintrin.h:37,
                 from crypto/sha256_sse41.cpp:8:
/usr/lib/gcc/x86_64-linux-gnu/13/include/tmmintrin.h: In function '__m128i sha256d64_sse41::{anonymous}::Read4(const unsigned char*, int)':
/usr/lib/gcc/x86_64-linux-gnu/13/include/tmmintrin.h:136:1: error: inlining failed in call to 'always_inline' '__m128i _mm_shuffle_epi8(__m128i, __m128i)': target specific option mismatch
  136 | _mm_shuffle_epi8 (__m128i __X, __m128i __Y)
      | ^~~~~~~~~~~~~~~~
crypto/sha256_sse41.cpp:55:28: note: called from here
   55 |     return _mm_shuffle_epi8(ret, _mm_set_epi32(0x0C0D0E0FUL, 0x08090A0BUL, 0x04050607UL, 0x00010203UL));
      |            ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-linux-gnu/13/include/tmmintrin.h:136:1: error: inlining failed in call to 'always_inline' '__m128i _mm_shuffle_epi8(__m128i, __m128i)': target specific option mismatch
  136 | _mm_shuffle_epi8 (__m128i __X, __m128i __Y)
      | ^~~~~~~~~~~~~~~~
crypto/sha256_sse41.cpp:55:28: note: called from here
   55 |     return _mm_shuffle_epi8(ret, _mm_set_epi32(0x0C0D0E0FUL, 0x08090A0BUL, 0x04050607UL, 0x00010203UL));
      |            ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:14520: crypto/libbitcoin_crypto_sse41_la-sha256_sse41.lo] Error 1
```

Both of these can / should be fixed up.

--- a/configure.ac
+++ b/configure.ac
@@ -504,11 +504,12 @@ TEMP_CXXFLAGS="$CXXFLAGS"
 CXXFLAGS="$SSE41_CXXFLAGS $CXXFLAGS"
 AC_MSG_CHECKING([for SSE4.1 intrinsics])
 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
-    #include <stdint.h>
     #include <immintrin.h>
   ]],[[
-    __m128i l = _mm_set1_epi32(0);
-    return _mm_extract_epi32(l, 3);
+    __m128i a = _mm_set1_epi32(0);
+    __m128i b = _mm_set1_epi32(1);
+    __m128i r = _mm_blend_epi16(a, b, 0xFF);
+    return _mm_extract_epi32(r, 3);
   ]])],
  [ AC_MSG_RESULT([yes]); enable_sse41=yes; AC_DEFINE([ENABLE_SSE41], [1], [Define this symbol to build code that uses SSE4.1 intrinsics]) ],
  [ AC_MSG_RESULT([no])]

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -53,14 +53,14 @@ LIBBITCOIN_CRYPTO = $(LIBBITCOIN_CRYPTO_BASE)
 if ENABLE_SSE41
 LIBBITCOIN_CRYPTO_SSE41 = crypto/libbitcoin_crypto_sse41.la
 LIBBITCOIN_CRYPTO += $(LIBBITCOIN_CRYPTO_SSE41)
+if ENABLE_X86_SHANI
+LIBBITCOIN_CRYPTO_X86_SHANI = crypto/libbitcoin_crypto_x86_shani.la
+LIBBITCOIN_CRYPTO += $(LIBBITCOIN_CRYPTO_X86_SHANI)
+endif
 endif
 if ENABLE_AVX2
 LIBBITCOIN_CRYPTO_AVX2 = crypto/libbitcoin_crypto_avx2.la
 LIBBITCOIN_CRYPTO += $(LIBBITCOIN_CRYPTO_AVX2)
-endif
-if ENABLE_X86_SHANI
-LIBBITCOIN_CRYPTO_X86_SHANI = crypto/libbitcoin_crypto_x86_shani.la
-LIBBITCOIN_CRYPTO += $(LIBBITCOIN_CRYPTO_X86_SHANI)
 endif
 if ENABLE_ARM_SHANI
 LIBBITCOIN_CRYPTO_ARM_SHANI = crypto/libbitcoin_crypto_arm_shani.la
@@ -612,7 +612,7 @@ crypto_libbitcoin_crypto_x86_shani_la_LDFLAGS = $(AM_LDFLAGS) -static
 crypto_libbitcoin_crypto_x86_shani_la_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS) -static
 crypto_libbitcoin_crypto_x86_shani_la_CPPFLAGS = $(AM_CPPFLAGS)
 crypto_libbitcoin_crypto_x86_shani_la_CXXFLAGS += $(X86_SHANI_CXXFLAGS)
-crypto_libbitcoin_crypto_x86_shani_la_CPPFLAGS += -DENABLE_X86_SHANI
+crypto_libbitcoin_crypto_x86_shani_la_CPPFLAGS += -DENABLE_SSE41 -DENABLE_X86_SHANI
 crypto_libbitcoin_crypto_x86_shani_la_SOURCES = crypto/sha256_x86_shani.cpp
 
 # See explanation for -static in crypto_libbitcoin_crypto_base_la's LDFLAGS and

--- a/src/crypto/sha256.cpp
+++ b/src/crypto/sha256.cpp
@@ -623,7 +623,7 @@ std::string SHA256AutoDetect(sha256_implementation::UseImplementation use_implem
         }
     }
 
-#if defined(ENABLE_X86_SHANI)
+#if defined(ENABLE_SSE41) && defined(ENABLE_X86_SHANI)
     if (have_x86_shani) {
         Transform = sha256_x86_shani::Transform;
         TransformD64 = TransformD64Wrapper<sha256_x86_shani::Transform>;

--- a/src/crypto/sha256_x86_shani.cpp
+++ b/src/crypto/sha256_x86_shani.cpp
@@ -6,7 +6,7 @@
 // Written and placed in public domain by Jeffrey Walton.
 // Based on code from Intel, and by Sean Gulley for the miTLS project.
 
-#ifdef ENABLE_X86_SHANI
+#if defined(ENABLE_SSE41) && defined(ENABLE_X86_SHANI)
 
 #include <stdint.h>
 #include <immintrin.h>

